org.eclipse.winery.repository/src/test/java/org/eclipse/winery/repository/targetallocation/criteria/minexternalconnections/MinExternalConnectionsTest.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+package org.eclipse.winery.repository.targetallocation.criteria.minexternalconnections;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.winery.common.ids.definitions.ServiceTemplateId;
+import org.eclipse.winery.model.tosca.TTopologyTemplate;
+import org.eclipse.winery.model.tosca.utils.ModelUtilities;
+import org.eclipse.winery.repository.TestWithGitBackedRepository;
+import org.eclipse.winery.repository.targetallocation.Criteria;
+import org.eclipse.winery.repository.targetallocation.util.AllocationUtils;
+import org.eclipse.winery.repository.targetallocation.util.TopologyWrapper;
+
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MinExternalConnectionsTest extends TestWithGitBackedRepository {
+
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(MinExternalConnectionsTest.class);
+
+    @BeforeEach
+    public void setUp() throws GitAPIException {
+        setRevisionTo("bb9ff7e08f7b72d30a2fae326740ef8051701671");
+    }
+
+    @Test
+    public void allocate1() throws Exception {
```

Comment:
Please rename the test to cleraly descripe what it does. Otherwiese it is more difficult to understand it.